platform-sdk/swirlds-common/src/main/java/com/swirlds/common/config/sources/MappedConfigSource.java

```diff
@@ -0,0 +1,143 @@
+ *
+ * @see ConfigSource
+ * @see ConfigMapping
+ */
+public class MappedConfigSource extends AbstractConfigSource {
+    private static final String PROPERTY_NOT_FOUND = "Property '%s' not found in original config source";
+    private static final String PROPERTY_ALREADY_DEFINED = "Property '%s' already defined";
+    private static final String DUPLICATE_PROPERTY = "Property '%s' already found in original config source";
+    private static final String PROPERTY_ALREADY_MAPPED = "Property '%s' has already a mapping defined";
+    private static final Logger logger = LogManager.getLogger(MappedConfigSource.class);
+
+    private final ConfigSource wrappedSource;
+
+    private final Queue<ConfigMapping> configMappings;
+    private final Map<String, String> properties;
+
+    /**
+     * Constructor that takes the wrapped config.
+     *
+     * @param wrappedSource the wrapped config
+     */
+    public MappedConfigSource(@NonNull final ConfigSource wrappedSource) {
+        this.wrappedSource = ArgumentUtils.throwArgNull(wrappedSource, "wrappedSource");
+        configMappings = new ConcurrentLinkedQueue<>();
+        properties = new HashMap<>();
+    }
+
+    /**
+     * Adds the mappedName {@code 'mappedName'<->'originalName'}
+     *
+     * @param mappedName   the mappedName name
+     * @param originalName the original name
+     */
+    public void addMapping(@NonNull final String mappedName, @NonNull final String originalName) {
+        addMapping(new ConfigMapping(mappedName, originalName));
+    }
+
+    /**
+     * Adds the mappedName {@code 'mappedName'<->'originalName'}
+     *
+     * @param configMapping defined mapping
+     */
+    public void addMapping(@NonNull final ConfigMapping configMapping) {
+        ArgumentUtils.throwArgNull(configMapping, "configMapping");
+
+        if (configMappings.stream()
+                .map(ConfigMapping::mappedName)
+                .anyMatch(m -> Objects.equals(m, configMapping.mappedName()))) {
+            throw new IllegalArgumentException(PROPERTY_ALREADY_DEFINED.formatted(configMapping.mappedName()));
+        }
+
+        if (configMappings.stream()
+                .map(ConfigMapping::originalName)
+                .anyMatch(o -> Objects.equals(o, configMapping.originalName()))) {
+            throw new IllegalArgumentException(PROPERTY_ALREADY_MAPPED.formatted(configMapping.originalName()));
+        }
+
+        configMappings.add(configMapping);
+    }
+
+    @Override
+    @NonNull
+    protected Map<String, String> getInternalProperties() {
+        if (properties.isEmpty()) {
+            final Map<String, String> internalProperties = wrappedSource.getProperties();
+            final Map<String, String> mappedProperties = new HashMap<>();
+
+            configMappings.forEach(configMapping -> {
+                if (internalProperties.containsKey(configMapping.mappedName())) {
+                    logger.info(DUPLICATE_PROPERTY.formatted(configMapping.mappedName()));
```

Comment:
Do these need to be exceptions or logged as warnings?   Does a LogMarker need to be added?

Reply:
I would not throw an exception it that case since it should not have a negative impact. It just do not really make sense. Regarding LogMarker: I would not introduce any new log marker that is not needed until we started https://github.com/hashgraph/hedera-services/issues/5424

Reply:
... but maybe we can switch the level to warn...

Reply:
please do not format string for the logger. use the `{}` placeholder that are supported by the logger instead.

Reply:
done